Handle polarity flag in pca9685 driver and remove unimplement/unused invert property

franckduriez opened the pull request:
Handle polarity flag in pca9685 driver and remove unimplement/unused invert property

franckduriez committed:
driver: pwm/pca9685: handle POLARITY flag

Handle polarity flag in pca9685 driver

Signed-off-by: Franck Duriez <franck.lucien.duriez@gmail.com>

franckduriez commented:
By the way, before this patch 
https://github.com/zephyrproject-rtos/zephyr/pull/94187/files#diff-72bf7fb30eae9d28a99069b01790c69b2bb053fb926cd008ea696b6b6b398523L26
is wrong, it should be 
```
pwms = <&pca9685_adafruit_pca9685 0 PWM_MSEC(20)>;
```
instead of
```
pwms = <&pca9685_adafruit_pca9685 0 PWM_MSEC(20) PWM_POLARITY_NORMAL>;
```
It only works because `PWM_POLARITY_NORMAL == 0` and if you change it by `PWM_POLARITY_INVERTED`, you get bizarre compilation errors in other parts of the device tree.

I used this file as base for my own board device tree and it took me long time to understand why I could not use `PWM_POLARITY_INVERTED`

kartben commented:
Changes probably need to be documented in the migration guide for current users of this binding

franckduriez committed:
driver: pwm/pca9685: remove unimplemented misleading property

Invert property was not implemented and is now redundant with the flag
pwm-cell which is more flexible in case you want some IO to be inverted
and some not

Signed-off-by: Franck Duriez <franck.lucien.duriez@gmail.com>

# Conflicts:
#	doc/releases/migration-guide-4.3.rst
